api/src/wfl/stage.clj

```diff
@@ -0,0 +1,30 @@
+(ns wfl.stage
+  "An interface for operations on a queue-based pipeline processing stage,
+  e.g. source, executor, or sink."
+  (:require [wfl.util :as util])
+  (:import [wfl.util UserException]))
+
+(defmulti validate-or-throw
+  "Validate the `request` request."
+  (fn [request] (:name request)))
+
+(defmethod validate-or-throw :default
+  [{:keys [name] :as request}]
+  (throw (UserException. "Invalid request - unknown name"
+                         (util/make-map name request))))
+
+(defmulti queue-peek
```

Comment:
I preferred peek-queue and so on. queue-peek just feels... backwards?

Reply:
Fair point.  `queue-*` naming convention had the benefit for me of grouping within autocomplete, but I can revert if the former is still your preference.